lib/ecto/adapters/sql.ex

```diff
@@ -355,6 +355,28 @@ defmodule Ecto.Adapters.SQL do
     opts
   end
 
+  @doc """
+  Check if the given `table` exists in the given `database`.
+
+  Returns `true` if the `table` exists in the `database` using the given
+  `adapter_meta`, otherwise `false`.
+  """
+  @spec table_exists?(Ecto.Repo.t | Ecto.Adapter.adapter_meta, table :: String.t, schema :: String.t) :: boolean
+  def table_exists?(repo_or_adapter, table, schema) do
+    query = """
+	SELECT 1
+      FROM information_schema.tables
+      WHERE table_name = '#{table}'
+      AND table_schema = '#{schema}';
```

Comment:
This will open us up to SQL injection attacks. Maybe you should write a regular Ecto.Query.